tools/semgrep.yml

```diff
@@ -166,3 +166,18 @@ rules:
             ...
     message: "Use change_user_is_active to mutate user_profile.is_active"
     severity: ERROR
+
+  - id: confirmation-object-get
+    languages: [python]
+    patterns:
+      - pattern-either:
+          - pattern: Confirmation.objects.get(...)
+          - pattern: Confirmation.objects.filter(..., confirmation_key=..., ...)
+      - pattern-not-inside: |
+          def get_object_from_key(...):
+            ...
+    paths:
+      exclude:
+        - zerver/tests/
```

Comment:
We can  probably exclude tests,  because it may be sensible to make more arbitrary queries in them, and should be harmless  too

Reply:
Yeah, excluding tests is very reasonable.